File: src/at/htlwels/bhit/wollersbergerjulian/Apfelmaenchen/model/Bereich.java
package at.htlwels.bhit.wollersbergerjulian.Apfelmaenchen.model;

// Created by julian on 11.07.17.
/**
 * Diese Klasse enthält die Grenzen eines
 * {@link at.htlwels.bhit.wollersbergerjulian.Apfelmaenchen.anzeigen.KomplexesCanvas}:
 * <br>minR, maxR, minI, maxI,
 */
public class Bereich {

    /* Der Bereich, in dem die komlexen Werte sind. */
    private double minR;
    private double maxR;
    private double minI;
    private double maxI;

    public Bereich(double minR, double maxR, double minI, double maxI) {
        this.minR = minR;
        this.maxR = maxR;
        this.minI = minI;
        this.maxI = maxI;
    }

    @Override
    public String toString() {
        return "Bereich{" +
                "minR=" + minR +
                ", maxR=" + maxR +
                ", minI=" + minI +
                ", maxI=" + maxI +
                '}';
    }

    public double getMinR() {
        return minR;
    }
    public void setMinR(double minR) {
        this.minR = minR;
    }
    public double getMaxR() {
        return maxR;
    }
    public void setMaxR(double maxR) {
        this.maxR = maxR;
    }
    public double getMinI() {
        return minI;
    }
    public void setMinI(double minI) {
        this.minI = minI;
    }
    public double getMaxI() {
        return maxI;
    }
    public void setMaxI(double maxI) {
        this.maxI = maxI;
    }

}

File: src/at/htlwels/bhit/wollersbergerjulian/apfelmännchen/model/Bereich.kt
package at.htlwels.bhit.wollersbergerjulian.apfelmännchen.model

import com.sun.jndi.ldap.Ber

// Created by julian on 11.07.17.
/**
 * Diese Klasse enthält die Grenzen eines
 * kartesischen Koordinatensystems.
 *(kxMin, kxMax, kyMin, kyMax)
 *
 * Außerdem werden die Größen kxSpanne
 * und kySpanne, also wie weit es von
 * Min bis Max ist, ausgerechnet.
 * (Ende minus Anfang.)
 */
data class Bereich(
        val kxMin: Double, val kxMax: Double, val kyMin: Double, val kyMax: Double) {

    /** Betrag; Ende minus Anfang. */
    val kxSpanne = kxMax - kxMin
    /** Betrag; Ende minus Anfang. */
    val kySpanne = kyMax - kyMin

    /** Erzeugt einen skalierten Bereich, bei dem alle Grenzen
     * einfach mit dem übergebenen Faktor multipliziert sind.
     * Weil Mathematik sind auch die Spannen um den selben
     * Faktor skaliert. */
    fun skalliert(zoomFaktor: Double): Bereich {
        return Bereich(kxMin*zoomFaktor, kxMax*zoomFaktor,
                kyMin*zoomFaktor, kyMax*zoomFaktor)
    }

    /** Addiert zu beiden kx deltaKX und zu beiden ky deltaKY. */
    fun verschoben(deltaKX: Double, deltaKY: Double): Bereich {
        return Bereich(kxMin+deltaKX, kxMax+deltaKX, kyMin+deltaKY, kyMax+deltaKY)
    }
}
